applications/asset_tracker/src/main.c

```diff
@@ -948,6 +948,36 @@ static void modem_configure(void)
 #endif
 }
 
+/**@brief Initializes the accelerometer device and
+ * configures trigger if set.
+ */
+static void accelerometer_init(void)
+{
+	struct device *accel_dev =
+	device_get_binding(CONFIG_ACCEL_DEV_NAME);
+
+	if (accel_dev == NULL) {
+		printk("Could not get %s device\n", CONFIG_ACCEL_DEV_NAME);
+		return;
+	}
+
+	struct sensor_trigger sensor_trig = {
+		.type = SENSOR_TRIG_THRESHOLD,
+	};
+
+	if (!IS_ENABLED(CONFIG_FLIP_POLL) && IS_ENABLED(CONFIG_ACCEL_USE_EXTERNAL)) {
```

Comment:
What do we want this function to do when this if-statement is false?

Is it necessary to get the device binding or declare sensor_trig?

I suspect not, and that we should move the entire function body inside of this if-statement.

Reply:
sensor_trig isn't required to be declared, but the rest of the application might be interested in getting the device binding still, even though we don't declare trigger handling.

I'll add the sensor_trig declaration inside the if-statement.

Reply:
>  the rest of the application might be interested in getting the device binding still

Right, but we have assigned it to a variable declared inside of the function, so they won't be able to access it.

Reply:
You are absolutely right. My mistake, sorry.